src/dev/flang/fuir/analysis/dfa/ValueSet.java

```diff
@@ -153,6 +153,22 @@ public Value joinInstances(Value v)
   }
 
 
+  /**
+   * Box this value. This works both for Instances as well as for value types
+   * such as i32, bool, etc.
+   */
+  Value box(DFA dfa, int vc, int rc, Context context)
+  {
+    Value result = null;
+    for (var v : _components.values())
+      {
+        var u = v.box(dfa, vc, rc, context);
+        result = result == null ? u : new ValueSet(result, u);
```

Comment:
This seems greater than O(n) but I think it could be O(n)?
Something like: 
```
_components
  .values()
  .flatMap(v -> v.box(...)._components.values())
 ```

Reply:
I have added a NYI-comment for now. There are several things in the DFA that have quadratic, maybe even cubic performance, these will have to be solved later. Very likely, the size of ValueSet would be limited to something like 10, which magically turns this code from O(n²) to O(1) since n<=10.